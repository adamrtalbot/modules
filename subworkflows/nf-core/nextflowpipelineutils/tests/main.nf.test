nextflow_workflow {

    name "Test Workflow NEXTFLOW_PIPELINE_UTILS"
    script "subworkflows/nf-core/nextflowpipelineutils/main.nf"
    workflow "NEXTFLOW_PIPELINE_UTILS"

    test("Should run no inputs") {

        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = null
                input[3] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should print version") {

        when {
            params {
                outdir = "tests/results"
            }
            workflow {
                """
                input[0] = true
                input[1] = false
                input[2] = null
                input[3] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }


    test("Should dump params") {

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = false
                input[1] = true
                input[2] = params.outdir
                input[3] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should not create params JSON if no output directory") {

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = false
                input[1] = true
                input[2] = null
                input[3] = false
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow, path(params.outdir).list()).match()
        }

    }

    test("Should check conda channels") {

        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = null
                input[3] = true
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

}
