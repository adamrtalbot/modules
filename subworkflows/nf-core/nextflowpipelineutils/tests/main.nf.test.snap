{
    "Should not create params JSON if no output directory": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            },
            [
                
            ]
        ],
        "timestamp": "2023-10-12T16:49:36.721659"
    },
    "Should print version": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-10-12T16:49:33.023807"
    },
    "Should dump params": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-10-12T16:53:21.862561"
    },
    "Should run no inputs": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-10-12T16:49:31.177792"
    },
    "Should check conda channels": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-10-12T16:49:38.736085"
    }
}